src/x11_utils.rs

```diff
@@ -37,6 +38,18 @@ pub trait TryParse: Sized {
     fn try_parse(value: &[u8]) -> Result<(Self, &[u8]), ParseError>;
 }
 
+/// A type implementing this trait can be parsed from some raw bytes and a list of fds.
+pub trait TryParseFd: Sized {
+    /// Try to parse the given values into an instance of this type.
+    ///
+    /// If parsing is successful, an instance of the type and a slice for the remaining data should
+    /// be returned. Otherwise, an error is returned.
```

Comment:
This doc is just copied from `TryParse`. It misses something about `fds`. How about something like this:
```
File descriptors are consumed by removing them from the beginning of the given `fds` `Vec`. If a file descriptor
is expected, but missing, a `ParseError` should be returned. If more file descriptors are provided than expected,
this is not an error and the remaining descriptors should be left in the `Vec`.
```
(Feel free to come up with a better formulation. I didn't even re-read what I just wrote.)

Reply:
sgtm. Added.